src/pages/posts/DetailsPost.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div class="page-post q-pa-lg">
+    <div class="col-6">
+      <q-img class="page-post__image" :src="detailsPost.image" />
+      <q-btn color="grey-4" text-color="brand" glossy unelevated icon="camera_enhance" label="Editar" @click="isEditImage = true" />
+    </div>
+    <div v-if="!this.$route.query.id || isEditImage">
+      <q-input class="col-3 q-mt-md" rounded outlined v-model="detailsPost.image" label="URL da imagem" />
+    </div>
+    <div class="row items-start justify-between">
+      <q-input class="col-3 q-mt-md" rounded outlined v-model="detailsPost.category" label="Categoria" />
+      <q-input class="col-4 q-mt-md" rounded outlined v-model="detailsPost.title" label="Título" />
+      <q-input class="col-3 q-mt-md" rounded outlined v-model="detailsPost.author" label="Autor" />
+      <q-input class="col-12 q-mt-md" rounded outlined v-model="detailsPost.description" label="Descrição" />
+    </div>
+
+    <div class="q-mt-md row justify-end q-pa-sm">
+      <q-btn color="brand q-mr-sm" label="Voltar" @click="$router.go(-1)"/>
+      <q-btn color="brand q-mr-sm" label="Salvar" @click="handleSave"/>
+      <q-btn v-if="this.$route.query.id" color="brand" label="Excluir" @click="isOpenModal = true"/>
```

Comment:
não precisa do `this` na view, porém é bom vc criar uma computada pra isso, pra validar as chaves e não correr o risco de crashar a página